doc/docs/master/how-tos/developer-workflow/build-pipelines.md

```diff
@@ -0,0 +1,118 @@
+# Build Pipelines
+
+A build pipeline is a useful feature when iterating on the code in your pipeline. They allow you to bypass the Docker build process and submit your code directly to the pipeline. In essence, build pipelines automate Steps 2-4 of the [pipeline workflow](working-with-pipelines.md). A diagram of the build pipeline process is shown below.
+
+![Developer workflow](../../assets/images/d_steps_build_pipeline.svg)
+
+!!! note
+      Build pipelines are not currently supported in Pachyderm Hub.
+
+Functionally, the build pipeline relies on a base Docker image that remains unchanged during the development process. Any code written for the pipeline is copied into this container, installs requirements and updates the code in the pipeline before running it.
+
+Using this feature requires a modification of the pipeline spec's `transform` object with options for the following fields:
+
+- `path`: An optional string specifying where the source code is relative to the pipeline spec path (or the cwd if the pipeline is fed into `pachctl` via stdin.)
+- `language`: An optional string specifying what language builder to use (see below.) Only works with official builders. If unspecified, `image` will be used instead.
+- `image`: An optional string specifying what builder image to use, if a non-official builder is desired. If unspecified, the `transform` object's `image` will be used instead.
+
+Below is a Python example of a build pipline.
+
+```json
+{
+  "pipeline": {
+    "name": "map"
+  },
+  "description": "A pipeline that tokenizes scraped pages and appends counts of words to corresponding files.",
+  "transform": {
+    "build": {
+      "language": "python",
+      "path": "./source"
+    }
+  },
+  "input": {
+    "pfs": {
+      "repo": "scraper",
+      "glob": "/*"
+    }
+  }
+}
+```
+
+A build pipeline can be submitted the same way as any other pipeline, for example:
+
+```bash
+pachctl update pipeline -f <pipeline name>
+```
+
+## How it works
+
+When submitted, the following actions occur:
+
+1. All files (code, etc.) are copied from the build path to a PFS repository, `<pipeline name>_build`, which we can think of as the source code repository. In the case above, everything in `./source` would be copied to to the PFS `<pipeline name>_build` repository.
+
+1. A pipeline called `<pipeline name>_build` starts, reading reading files from the `<pipeline name>_build` repo and runs the `build.sh` script to pull dependencies and compile any requirements from the source code.
+
+1. The running pipeline, `<pipeline name>`, is updated (or created if it doesn't already exist) with the the new source files and built assets.
+
+!!! note
+      You can optionally specify a `.pachignore` file, which uses [ohmyglob](https://github.com/pachyderm/ohmyglob) entries to prevent certain files from getting pushed to this repo.
```

Comment:
```suggestion
      You can optionally specify a `.pachignore` file in the source root directory, which uses [ohmyglob](https://github.com/pachyderm/ohmyglob) entries to prevent certain files from getting pushed to this repo.
```